### ISSUE TITLE
Q: What is the recommended display format for SingleOrNoValue options?

### ISSUE THREAD

**Issue by bjorg** [natemcmaster/CommandLineUtils#113]
I was wondering what the recommended format is for describing `SingleOrNoValue` type options.

My first attempt was to use `[:<VALUE>]` to indicate that the suffix is optional, but that causes a parse exceptions.
```csharp
var verboseLevelOption = command.Option(
  "-V|--verbose[:<LEVEL>]",
   "(optional) Show verbose output (0=quiet, 1=normal, 2=detailed, 3=exceptions)",
   CommandOptionType.SingleOrNoValue
);
```

Error:
```
Unhandled Exception: System.ArgumentException: Invalid template pattern '-V|--verbose[:<LEVEL>]'
```

For now, I settled on the following, which I feel is misleading since it implies the suffix is required.
```csharp
var verboseLevelOption = command.Option(
  "-V|--verbose:<LEVEL>",
  "(optional) Show verbose output (0=quiet, 1=normal, 2=detailed, 3=exceptions)",
  CommandOptionType.SingleOrNoValue
);
```

Thanks for clarifying!

**Comment by natemcmaster** [comment]
I like your suggestion for `-V|--verbose[:<LEVEL>]`. I'd take a PR to improve the parsing to handle this syntax.

**Comment by bjorg** [comment]
Will do. Any unit tests I should run/adapt?

**Comment by natemcmaster** [comment]
Obviously make sure existing unit tests don't begin to fail, but otherwise, I think https://github.com/natemcmaster/CommandLineUtils/blob/master/test/CommandLineUtils.Tests/CommandOptionTests.cs is probably the best place to put this.

**Comment by natemcmaster** [comment]
Fixed with #117
